hilti/toolchain/include/ast/builder/builder.h

```diff
@@ -509,6 +509,9 @@ class Builder : public builder::NodeFactory {
         block()->_add(context(), local(std::move(id), t, std::move(args), std::move(m)));
     }
 
+    // Adds a temporary to store an attribute's expression, if necessary.
+    Expression* addAttributeTmp(Attribute* attr, const std::string& prefix);
```

Comment:
Mostly personal taste  but I find the "add...Tmp" potentially misleading given that the function might end up *not* adding a tmp (for constants). `evaluateAttributeExpression` maybe? (Fine to leave as is if you prefer.)

Reply:
Sure. `maybeAddAttributeTmp` wasn't catchy :)